src/Components/Common/ErrorPopup/ErrorPopup.tsx

```diff
@@ -0,0 +1,53 @@
+import { type FC, type ReactElement, type MouseEvent, type KeyboardEvent, useEffect, useRef } from 'react';
+import classes from './ErrorPopup.module.scss';
+import { Error } from '../Error/Error';
+import type { ErrorType, Nullable } from '../../../utils/types/common';
+import { KeyboardEventCode } from '../../../utils/types/enums';
+
+type PropsType = {
+  errorObject: Nullable<ErrorType>;
+  resetError: (error: Nullable<ErrorType>) => void;
+};
+
+export const ErrorPopup: FC<PropsType> = ({ errorObject, resetError }): ReactElement | null => {
+  const errorPopup = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    errorPopup.current?.focus();
+  }, [errorObject]);
+
+  const onButtonClick = (): void => {
+    resetError(null);
+  };
+
+  const onPopupMouseClick = (event: MouseEvent<HTMLDivElement>): void => {
+    if ((event.target as Element).className === errorPopup.current?.className) {
+      resetError(null);
```

Comment:
Will it reset error in case user click on any place inside the popup?

Reply:
No, it will not.

The error will be reseted in 3 cases:
1. If user clicks on the "Close" button inside the popup;
2. If user clicks on any place outside the popup;
3. If user presses the "Escape" key.